lib/handlers/ntlm.ts

```diff
@@ -2,124 +2,160 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 import ifm = require('../Interfaces');
-
 import http = require("http");
 import https = require("https");
+
 var _ = require("underscore");
 var ntlm = require("../opensource/node-http-ntlm/ntlm");
 
+interface INtlmOptions {
+    username?: string,
+    password?: string,
+    domain: string,
+    workstation: string
+}
+
 export class NtlmCredentialHandler implements ifm.IRequestHandler {
-    username: string;
-    password: string;
-    workstation: string;
-    domain: string;
-
-    constructor(username: string, password: string,  domain?: string, workstation?: string) {
-        this.username = username;
-        this.password = password;
+    private _ntlmOptions: INtlmOptions;
+
+    constructor(username: string, password: string, workstation?: string, domain?: string) {
+        this._ntlmOptions = <INtlmOptions>{};
+
+        this._ntlmOptions.username = username;
+        this._ntlmOptions.password = password;
+
         if (domain !== undefined) {
-            this.domain = domain;
+            this._ntlmOptions.domain = domain;
         }
         if (workstation !== undefined) {
-            this.workstation = workstation;
+            this._ntlmOptions.workstation = workstation;
         }
     }
 
-    prepareRequest(options:any): void {
+    public prepareRequest(options: http.RequestOptions): void {
         // No headers or options need to be set.  We keep the credentials on the handler itself.
         // If a (proxy) agent is set, remove it as we don't support proxy for NTLM at this time
         if (options.agent) {
             delete options.agent;
         }
     }
 
-    canHandleAuthentication(res: ifm.IHttpResponse): boolean {
-        if (res && res.statusCode === 401) {
+    public canHandleAuthentication(response: ifm.IHttpClientResponse): boolean {
+        if (response && response.message && response.message.statusCode === 401) {
             // Ensure that we're talking NTLM here
             // Once we have the www-authenticate header, split it so we can ensure we can talk NTLM
-            var wwwAuthenticate = res.headers['www-authenticate'];
-            if (wwwAuthenticate !== undefined) {
-                var mechanisms = wwwAuthenticate.split(', ');
-                var idx =  mechanisms.indexOf("NTLM");
-                if (idx >= 0) {
+            const wwwAuthenticate = response.message.headers['www-authenticate'];
+
+            if (wwwAuthenticate) {
+                const mechanisms = wwwAuthenticate.split(', ');
```

Comment:
is this safe or should we split on comma and then trim and if not empty, then proceed

Reply:
According to [the spec](https://tools.ietf.org/html/rfc7235#section-4.1) it looks like there will always be a space after commas.